[BUG] Malformed template in workflow crashes consumption

### Description

I used incorrect syntax in a workflow's "Assign title" action that crashed consumption with a template warning. 

### Steps to reproduce

1. Create a Workflow with a trigger on document consumption.
2. Add a "Assign title" action with this template (used Liquid-style filter syntax):
```
   {{ created }} - {{ correspondent | default: "Unknown" }}
```
3. Consume any document that matches the workflow's trigger.
4. Document fails to import entirely.


### Webserver logs

```bash
[WARNING] [paperless.templating] Template syntax error in title generation: expected token 'end of print statement', got ':'
[ERROR] [paperless.consumer] The following error occurred while storing document scan.pdf after parsing: 'NoneType' object is not subscriptable
Traceback (most recent call last):
  ...
  File "src/documents/signals/handlers.py", line 867, in run_workflows
    document.title = document.title[:128]
                     ~~~~~~~~~~~~~~^^^^^^
TypeError: 'NoneType' object is not subscriptable
```

### Browser logs

```bash

```

### Services logs

```bash

```

### Paperless-ngx version

2.20.15

### Host OS

Ubuntu 26.04 LTS

### Installation method

Docker - official image

### System status

```json
{
    "pngx_version": "2.20.15",
    "server_os": "Linux-7.0.0-27-generic-x86_64-with-glibc2.41",
    "install_type": "docker",
    "storage": {
        "total": 1805070348288,
        "available": 1447151128576
    },
    "database": {
        "type": "postgresql",
        "url": "paperless",
        "status": "OK",
        "error": null,
        "migration_status": {
            "latest_migration": "paperless_mail.0001_initial_squashed_0009_mailrule_assign_tags",
            "unapplied_migrations": []
        }
    },
    "tasks": {
        "redis_url": "redis://broker:6379",
        "redis_status": "OK",
        "redis_error": null,
        "celery_status": "OK",
        "celery_url": "celery@89a1f7c4f5b2",
        "celery_error": null,
        "index_status": "OK",
        "index_last_modified": "2026-07-08T16:49:15.024100-07:00",
        "index_error": null,
        "classifier_status": "OK",
        "classifier_last_trained": "2026-07-09T04:05:02.983417Z",
        "classifier_error": null,
        "sanity_check_status": "OK",
        "sanity_check_last_run": "2026-07-09T04:31:33.490788Z",
        "sanity_check_error": null
    },
    "websocket_connected": "OK"
}
```

### Browser

_No response_

### Configuration changes

_No response_

### Please confirm the following

- [x] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [x] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [x] I have already searched for relevant existing issues and discussions before opening this report.
- [x] I have updated the title field above with a concise description.

--- a/src/documents/tests/test_workflows.py
+++ b/src/documents/tests/test_workflows.py
@@ -1847,6 +1847,45 @@ class TestWorkflows(
 
         self.assertEqual(doc.title, "Doc {created_year]")
 
+    def test_document_added_malformed_title_template_falls_back(self) -> None:
+        """
+        GIVEN:
+            - Existing workflow with added trigger type
+            - Assign title field is malformed Jinja2 syntax
+        WHEN:
+            - File that matches is added
+        THEN:
+            - Title assignment is skipped and the original title is kept
+        """
+        trigger = WorkflowTrigger.objects.create(
+            type=WorkflowTrigger.WorkflowTriggerType.DOCUMENT_ADDED,
+            filter_filename="*sample*",
+        )
+        action = WorkflowAction.objects.create(
+            assign_title="Doc {{ unclosed",
+        )
+        w = Workflow.objects.create(
+            name="Workflow 1",
+            order=0,
+        )
+        w.triggers.add(trigger)
+        w.actions.add(action)
+        w.save()
+
+        doc = Document.objects.create(
+            original_filename="sample.pdf",
+            title="sample test",
+            content="Hello world bar",
+        )
+
+        document_consumption_finished.send(
+            sender=self.__class__,
+            document=doc,
+        )
+
+        doc.refresh_from_db()
+        self.assertEqual(doc.title, "sample test")
+
     def test_document_updated_workflow_ignores_version_documents(self) -> None:
         trigger = WorkflowTrigger.objects.create(
             type=WorkflowTrigger.WorkflowTriggerType.DOCUMENT_UPDATED,

--- a/src/documents/workflows/mutations.py
+++ b/src/documents/workflows/mutations.py
@@ -40,7 +40,7 @@ def apply_assignment_to_document(
 
     if action.assign_title:
         try:
-            document.title = parse_w_workflow_placeholders(
+            title = parse_w_workflow_placeholders(
                 action.assign_title,
                 document.correspondent.name if document.correspondent else "",
                 document.document_type.name if document.document_type else "",
@@ -53,6 +53,8 @@ def apply_assignment_to_document(
                 "",  # no urls in titles
                 document.pk,
             )
+            if title:
+                document.title = title
         except Exception:  # pragma: no cover
             logger.exception(
                 f"Error occurred parsing title assignment '{action.assign_title}', falling back to original",

